### PR TITLE
Assembly AI STT endpoint

### PR DESCRIPTION
**Pull request by chrisveilleux**
## Description
Adds a new endpoint to the Device API to use Assembly AI as a STT transcription service provider.

## How to test
Run the behavior tests.

## Contributor license agreement signed?
CLA [X]

### CODE REVIEW

File: api/public/tests/features/transcribe_audio.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Then the request will be successful
+    And Google's transcription will be correct
+    And the device's last contact time is updated
+
+  @stt
+  Scenario: Transcribe audio using Assembly AI
+    When Utterance "what time is it" is transcribed using Mycroft's transcription service
+    Then the request will be successful
+    And the transcription will be returned to the device
+    And the device's last contact time is updated
```
Comment (krisgesling) [inline api/public/tests/features/transcribe_audio.feature:16]: Would it be worth having an audio file with very clear and known speech in it so the behave test could cover the entire audio to response cycle?

**Review by krisgesling — APPROVED** [review]
Looks great!

File: api/public/tests/features/transcribe_audio.feature (hunk 1 above)
Comment (chrisveilleux) [reply api/public/tests/features/transcribe_audio.feature:16]: I am not sure I understand.  There is a .flac file used in the test.

Comment (krisgesling) [reply api/public/tests/features/transcribe_audio.feature:16]: oh sorry ignore me, I somehow missed the step definition and assumed the `when` step was returning a mock "what time is it" utterance.

**Pull request merged**